e2e/libs/utility/resource_cache.py

```diff
@@ -0,0 +1,79 @@
+from utility.constant import CACHE_NODE_ID_OF_REPLICA_0
+from utility.constant import CACHE_NODE_ID_OF_REPLICA_1
+from utility.constant import CACHE_NODE_ID_OF_REPLICA_2
+
+class ResourceCache(object):
```

Comment:
Cache is a class object (singleton), so it should support concurrency when we would like to run test cases in parallel.